nixops/deployment.py

```diff
@@ -1488,7 +1489,8 @@ def worker(m: nixops.resources.ResourceState) -> None:
                 m._errored = True
                 raise
             finally:
-                m._destroyed_event.set()
+                if dep._destroyed_event is not None:
```

Comment:
cc @grahamc  this triggers the following error

```
    if dep._destroyed_event is not None:
UnboundLocalError: local variable 'dep' referenced before assignment
```

Reply:
omg!

Reply:
How did you catch this error? and why didn't I? :/

Reply:
Randomly xD

Was playing around a deployment containing some missing resource

```
+-----------------------+---------+-------------------+---------------------------------------+------------+
| Name                  |  Status | Type              | Resource Id                           | IP address |
+-----------------------+---------+-------------------+---------------------------------------+------------+
| test                  |    Up   | command-output    | test-afc21edad3e1050cde1bde655014dd59 |            |
| firstGrafanaDashboard | Missing | grafana-dashboard |                                       |            |
+-----------------------+---------+-------------------+---------------------------------------+------------+
```
And when I tried to destroy all resources, came across that error :crossed_fingers: 

```
$ nixops destroy -d grafana-test  
warning: are you sure you want to destroy test? (y/N) y
test> destroying...
Traceback (most recent call last):
  File "/home/tewfikghariani/.cache/pypoetry/virtualenvs/nixops-grafana-FCCl2_6c-py3.7/bin/nixops", line 11, in <module>
    load_entry_point('nixops', 'console_scripts', 'nixops')()
  File "/home/tewfikghariani/.cache/pypoetry/virtualenvs/nixops-grafana-FCCl2_6c-py3.7/src/nixops/nixops/__main__.py", line 705, in main
    args.op(args)
  File "/home/tewfikghariani/.cache/pypoetry/virtualenvs/nixops-grafana-FCCl2_6c-py3.7/src/nixops/nixops/script_defs.py", line 632, in op_destroy
    include=args.include or [], exclude=args.exclude or [], wipe=args.wipe
  File "/home/tewfikghariani/.cache/pypoetry/virtualenvs/nixops-grafana-FCCl2_6c-py3.7/src/nixops/nixops/deployment.py", line 1506, in destroy_resources
    "destroy", lambda: self._destroy_resources(include, exclude, wipe)
  File "/home/tewfikghariani/.cache/pypoetry/virtualenvs/nixops-grafana-FCCl2_6c-py3.7/src/nixops/nixops/deployment.py", line 1366, in run_with_notify
    f()
  File "/home/tewfikghariani/.cache/pypoetry/virtualenvs/nixops-grafana-FCCl2_6c-py3.7/src/nixops/nixops/deployment.py", line 1506, in <lambda>
    "destroy", lambda: self._destroy_resources(include, exclude, wipe)
  File "/home/tewfikghariani/.cache/pypoetry/virtualenvs/nixops-grafana-FCCl2_6c-py3.7/src/nixops/nixops/deployment.py", line 1496, in _destroy_resources
    nr_workers=-1, tasks=list(self.resources.values()), worker_fun=worker
  File "/home/tewfikghariani/.cache/pypoetry/virtualenvs/nixops-grafana-FCCl2_6c-py3.7/src/nixops/nixops/parallel.py", line 106, in run_tasks
    raise list(exceptions.values())[0]
  File "/home/tewfikghariani/.cache/pypoetry/virtualenvs/nixops-grafana-FCCl2_6c-py3.7/src/nixops/nixops/parallel.py", line 70, in thread_fun
    work_result = (worker_fun(t), None, t.name)
  File "/home/tewfikghariani/.cache/pypoetry/virtualenvs/nixops-grafana-FCCl2_6c-py3.7/src/nixops/nixops/deployment.py", line 1492, in worker
    if dep._destroyed_event is not None:
UnboundLocalError: local variable 'dep' referenced before assignment
```

Reply:
Thanks, @tewfic-ghariani! Oops! I've opened up a fixup PR. Can you give it a test? https://github.com/NixOS/nixops/pull/1348

Reply:
Works fine now! Thanks